### PR TITLE
 upgrade substrate to 558a68f83

### PR DESCRIPTION
**Pull request by brenzi**

### CODE REVIEW

File: node/src/service.rs [hunk 1]
```diff
@@ -216,7 +225,7 @@ pub fn new_full(mut config: Configuration) -> Result<TaskManager, ServiceError>
 				deny_unsafe,
 			};
 
-			crate::rpc::create_full(deps)
+			Ok(crate::rpc::create_full(deps))
```
Comment (brenzi) [inline node/src/service.rs:228]: @clangenb I get 
```
error[E0277]: the trait bound `RuntimeApiImpl<sp_runtime::generic::Block<sp_runtime::generic::Header<u32, BlakeTwo256>, OpaqueExtrinsic>, sc_service::client::client::Client<sc_client_db::Backend<sp_runtime::generic::Block<sp_runtime::generic::Header<u32, BlakeTwo256>, OpaqueExtrinsic>>, LocalCallExecutor<sp_runtime::generic::Block<sp_runtime::generic::Header<u32, BlakeTwo256>, OpaqueExtrinsic>, sc_client_db::Backend<sp_runtime::generic::Block<sp_runtime::generic::Header<u32, BlakeTwo256>, OpaqueExtrinsic>>, NativeElseWasmExecutor<ExecutorDispatch>>, sp_runtime::generic::Block<sp_runtime::generic::Header<u32, BlakeTwo256>, OpaqueExtrinsic>, encointer_node_notee_runtime::RuntimeApi>>: pallet_encointer_communities_rpc_runtime_api::CommunitiesApi<sp_runtime::generic::Block<sp_runtime::generic::Header<u32, BlakeTwo256>, OpaqueExtrinsic>>` is not satisfied
   --> node/src/service.rs:228:7
    |
228 |             Ok(crate::rpc::create_full(deps))
    |                ^^^^^^^^^^^^^^^^^^^^^^^ the trait `pallet_encointer_communities_rpc_runtime_api::CommunitiesApi<sp_runtime::generic::Block<sp_runtime::generic::Header<u32, BlakeTwo256>, OpaqueExtrinsic>>` is not implemented for `RuntimeApiImpl<sp_runtime::generic::Block<sp_runtime::generic::Header<u32, BlakeTwo256>, OpaqueExtrinsic>, sc_service::client::client::Client<sc_client_db::Backend<sp_runtime::generic::Block<sp_runtime::generic::Header<u32, BlakeTwo256>, OpaqueExtrinsic>>, LocalCallExecutor<sp_runtime::generic::Block<sp_runtime::generic::Header<u32, BlakeTwo256>, OpaqueExtrinsic>, sc_client_db::Backend<sp_runtime::generic::Block<sp_runtime::generic::Header<u32, BlakeTwo256>, OpaqueExtrinsic>>, NativeElseWasmExecutor<ExecutorDispatch>>, sp_runtime::generic::Block<sp_runtime::generic::Header<u32, BlakeTwo256>, OpaqueExtrinsic>, encointer_node_notee_runtime::RuntimeApi>>`
    | 
   ::: node/src/rpc.rs:35:8
    |
35  | pub fn create_full<C, P, TBackend>(
    |        ----------- required by a bound in this
...
45  |     C::Api: pallet_encointer_communities_rpc_runtime_api::CommunitiesApi<Block>,
    |             ------------------------------------------------------------------- required by this bound in `create_full`

error[E0277]: the trait bound `RuntimeApiImpl<sp_runtime::generic::Block<sp_runtime::generic::Header<u32, BlakeTwo256>, OpaqueExtrinsic>, sc_service::client::client::Client<sc_client_db::Backend<sp_runtime::generic::Block<sp_runtime::generic::Header<u32, BlakeTwo256>, OpaqueExtrinsic>>, LocalCallExecutor<sp_runtime::generic::Block<sp_runtime::generic::Header<u32, BlakeTwo256>, OpaqueExtrinsic>, sc_client_db::Backend<sp_runtime::generic::Block<sp_runtime::generic::Header<u32, BlakeTwo256>, OpaqueExtrinsic>>, NativeElseWasmExecutor<ExecutorDispatch>>, sp_runtime::generic::Block<sp_runtime::generic::Header<u32, BlakeTwo256>, OpaqueExtrinsic>, encointer_node_notee_runtime::RuntimeApi>>: pallet_encointer_bazaar_rpc_runtime_api::BazaarApi<sp_runtime::generic::Block<sp_runtime::generic::Header<u32, BlakeTwo256>, OpaqueExtrinsic>, AccountId32>` is not satisfied
   --> node/src/service.rs:228:7
    |
228 |             Ok(crate::rpc::create_full(deps))
    |                ^^^^^^^^^^^^^^^^^^^^^^^ the trait `pallet_encointer_bazaar_rpc_runtime_api::BazaarApi<sp_runtime::generic::Block<sp_runtime::generic::Header<u32, BlakeTwo256>, OpaqueExtrinsic>, AccountId32>` is not implemented for `RuntimeApiImpl<sp_runtime::generic::Block<sp_runtime::generic::Header<u32, BlakeTwo256>, OpaqueExtrinsic>, sc_service::client::client::Client<sc_client_db::Backend<sp_runtime::generic::Block<sp_runtime::generic::Header<u32, BlakeTwo256>, OpaqueExtrinsic>>, LocalCallExecutor<sp_runtime::generic::Block<sp_runtime::generic::Header<u32, BlakeTwo256>, OpaqueExtrinsic>, sc_client_db::Backend<sp_runtime::generic::Block<sp_runtime::generic::Header<u32, BlakeTwo256>, OpaqueExtrinsic>>, NativeElseWasmExecutor<ExecutorDispatch>>, sp_runtime::generic::Block<sp_runtime::generic::Header<u32, BlakeTwo256>, OpaqueExtrinsic>, encointer_node_notee_runtime::RuntimeApi>>`
    | 
   ::: node/src/rpc.rs:35:8
    |
35  | pub fn create_full<C, P, TBackend>(
    |        ----------- required by a bound in this
...
46  |     C::Api: pallet_encointer_bazaar_rpc_runtime_api::BazaarApi<Block, AccountId>,
    |             -------------------------------------------------------------------- required by this bound in `create_full`


```
can't see what's different from before. Can you help?

File: runtime/src/lib.rs [hunk 2]
```diff
@@ -504,28 +519,28 @@ impl_runtime_apis! {
...
+			use frame_support::traits::StorageInfoTrait;
+			use frame_system_benchmarking::Pallet as SystemBench;
 
-		fn get_name(cid: &CommunityIdentifier) -> Option<PalletString> {
-			EncointerCommunities::get_name(cid)
-		}
-	}
+			let mut list = Vec::<BenchmarkList>::new();
 
-	impl encointer_bazaar_rpc_runtime_api::BazaarApi<Block, AccountId> for Runtime {
```
Comment (clangenb) [inline runtime/src/lib.rs]: @brenzi : Regarding your compiler error. This is because you removed the runtime api implementation here.

Comment (brenzi) [reply runtime/src/lib.rs]: aarghh. an unmindful copy-paste from upstream

File: runtime/src/lib.rs [hunk 3]
```diff
@@ -553,7 +588,7 @@ impl_runtime_apis! {
 			add_benchmark!(params, batches, frame_system, SystemBench::<Runtime>);
 			add_benchmark!(params, batches, pallet_balances, Balances);
 			add_benchmark!(params, batches, pallet_timestamp, Timestamp);
-			add_benchmark!(params, batches, encointer_communities, EncointerCommunities);
+			add_benchmark!(params, batches, pallet_template, TemplateModule);
```
Comment (clangenb) [inline runtime/src/lib.rs:592]: Was this intentional? If not, we probably also need to add the encointer_communities to the list_benchmarks above?

Comment (clangenb) [reply runtime/src/lib.rs:592]: @brenzi

Comment (brenzi) [reply runtime/src/lib.rs:592]: not intentional

Comment (brenzi) [reply runtime/src/lib.rs:592]: fixed in 672ed4e299935

**Pull request merged**